crates/common/tedge_utils/src/notify.rs

```diff
@@ -0,0 +1,393 @@
+                return events.contains(event);
+            }
+        }
+        false
+    }
+}
+
+pub struct NotifyStream {
+    watcher: INotifyWatcher,
+    pub rx: Receiver<(PathBuf, FileEvent)>,
+    metadata: WatchMetadata,
+}
+
+impl TryDefault for NotifyStream {
+    type Error = NotifyStreamError;
+
+    fn try_default() -> Result<Self, Self::Error> {
+        let (tx, rx) = channel(1024);
+
+        // Automatically select the best implementation for your platform.
+        // You can also access each implementation directly e.g. INotifyWatcher.
+        let watcher = RecommendedWatcher::new(
+            move |res: Result<notify::Event, notify::Error>| {
+                futures::executor::block_on(async {
+                    if let Ok(notify_event) = res {
+                        match notify_event.kind {
+                            EventKind::Modify(ModifyKind::Data(DataChange::Any)) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Modified)).await.unwrap();
+                                }
+                            }
+
+                            EventKind::Access(AccessKind::Close(AccessMode::Write)) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Modified)).await.unwrap();
+                                }
+                            }
+                            EventKind::Create(CreateKind::File) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Created)).await.unwrap();
+                                }
+                            }
+                            EventKind::Create(CreateKind::Folder) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Created)).await.unwrap();
+                                }
+                            }
+                            //EventKind::Modify(_) => todo!(),
+                            EventKind::Remove(RemoveKind::File) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Deleted)).await.unwrap();
+                                }
+                            }
+                            EventKind::Remove(RemoveKind::Folder) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Deleted)).await.unwrap();
+                                }
+                            }
+                            _other => {
+                                //dbg!(_other);
+                            }
+                        }
+                    }
+                })
+            },
+            Config::default().with_poll_interval(std::time::Duration::from_secs(2)),
```

Comment:
The duration can be defined as a constant.
```suggestion
            Config::default().with_poll_interval(POLL_INTERVAL),
   ```